test/unit/services/tracker.ts

```diff
@@ -148,6 +230,23 @@ suite('Tracker Service', ({ expect, spy, stub, itShouldExtendBaseService }) => {
       });
     });
 
+    it('should override event with the override function if it exists', () => {
+      const sendEvent = service.sendEvent = spy();
+      const currEvent = {
+        metadata,
+        search: { id, origin: { ...DEFAULT_ORIGINS, search: true }, },
+      };
+      const result = { a: 'b' };
+      const override = spy(() => result);
+      service.getMetadata = spy(() => metadata);
```

Comment:
I don't think this needs to be a `spy` if we're not making any assertions on it.